libraries/typings-generator/src/TypingsGenerator.ts

```diff
@@ -121,7 +126,35 @@ export class TypingsGenerator {
     });
   }
 
+  /**
+   * Register file dependencies that may effect the typings of a target file.
+   * Note: This feature is only useful in watch mode.
+   * The registerDependency method must be called in the body of parseAndGenerateTypings every
+   * time because the registry for a file is cleared at the beginning of processing.
+   */
+  public registerDependency(target: string, dependency: string): void {
+    if (!this._targetMap.has(target)) {
```

Comment:
```ts
let targetDependencyMap: Map<string, boolean> | undefined = this._targetMap.get(target);
if (!targetDependencyMap) {
  targetDependencyMap= new Map();
  this._targetMap.set(target, targetDependencyMap);
}
```

Do the same for the below map query. Reduces queries against the map

Reply:
Great idea. Done.